---
author: Ruihan Wang
title: "Variant Calling"
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
  md_document:
    variant: markdown_github
bibliography: bibliography.ris
---

## Methods
### 1
get the genome info from from ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/GRCh38.primary_assembly.genome.fa.gz and then etrieve reads in SRA (script getGenome.sh and getReads.sh)
### 2
Using Trimmomatict do Quality trim (script trimReads.sh)[@trim]
### 3
Index the genome for use by BWA (indexGenome.sh)[@index]
### 4
Align the reads using bwa mem (alignReads.sh)[@align]
### 5
Using samtool to sort the file (script sort.sh)[@sort]
### 6
Using samtool to index reads (script indexReads.sh)[@in]
### 7
Using DeepVariant to do a analysis (script runDeepVariant.sh) to generate a VCF file[@deep]

## Referenece 
