---
title: "Cancer Genomics"
output: html_notebook
---


## Installation
Install the maftools package
```{r}

if (!require("BiocManager"))
    install.packages("BiocManager")
BiocManager::install("maftools")
```
## Overview
### Reading the MAF files
```{r}
library(maftools)
laml.maf = system.file('extdata', 'tcga_laml.maf.gz', package = 'maftools') 
laml.clin = system.file('extdata', 'tcga_laml_annot.tsv', package = 'maftools') 

laml = read.maf(maf = laml.maf, clinicalData = laml.clin)
```

### Summarized MAF file is stored as an MAF object
```{r}
laml
getSampleSummary(laml)
```


```{r}
getFields(laml)
```


```{r}
getClinicalData(laml)
```


```{r}
getGeneSummary(laml)
```


```{r}
write.mafSummary(maf = laml, basename = 'laml')
```

## Visualization
### Plotting MAF summary.
```{r}
plotmafSummary(maf = laml, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```

### Drawing oncoplots

```{r}
oncoplot(maf = laml, top = 10)
```

### Drawing waterfall plots
```{r}
oncostrip(maf = laml, genes = c('DNMT3A','NPM1', 'RUNX1'))
```
### Transition and Transversions
titv function classifies SNPs into Transitions and Transversions and returns a list of summarized tables in various ways. Summarized data can also be visualized as a boxplot showing overall distribution of six different conversions and as a stacked barplot showing fraction of conversions in each sample.

```{r}
laml.titv = titv(maf = laml, plot = FALSE, useSyn = TRUE)
#plot titv summary
plotTiTv(res = laml.titv)
```


### Lollipop plots for amino acid changes
Showing mutation spots on protein structure with lollipop plots. It will show the most mutant spot.
```{r}
#lollipop plot for DNMT3A, which is one of the most frequent mutated gene in Leukemia.
lollipopPlot(maf = laml, gene = 'DNMT3A', AACol = 'Protein_Change', showMutationRate = TRUE)
```

### Labelling points
```{r}
lollipopPlot(maf = laml, gene = 'KIT', AACol = 'Protein_Change', labelPos = 816, refSeqID = 'NM_000222')
```


### Rainfall plots
Rainfall plot also highlights regions where potential changes in inter-event distances are located.
```{r}
brca <- system.file("extdata", "brca.maf.gz", package = "maftools")
brca = read.maf(maf = brca, verbose = FALSE)
rainfallPlot(maf = brca, detectChangePoints = TRUE, pointSize = 0.6)
```

### Compare mutation load against TCGA cohorts
TCGA contains over 30 different cancer cohorts and median mutation load across them varies from as low as 7 per exome to as high as 315 per exome. It is informative to see how mutation load in given maf stands against TCGA cohorts. 
```{r}
laml.mutload = tcgaCompare(maf = laml, cohortName = 'Example-LAML')

```

### Plotting VAF
```{r}
plotVaf(maf = laml, vafCol = 'i_TumorVAF_WU')

```


### Genecloud
Plot word cloud plot for mutated genes with the function geneCloud
```{r}
geneCloud(input = laml, minMut = 3)
```
## Analysis
### Somatic Interactions
Genes occurs mutated in cancer are co-occuring in their mutation pattern. using somaticInteractions to detect this relationship.

```{r}
somaticInteractions(maf = laml, top = 25, pvalue = c(0.05, 0.1))
```
### Detecting cancer driver genes based on positional clustering
Detecting cancer dirver genes using oncodrive function that using  oncodriveCLUST algorithm. 
```{r}
laml.sig = oncodrive(maf = laml, AACol = 'Protein_Change', minMut = 5, pvalMethod = 'zscore')
head(laml.sig)

```

```{r}
plotOncodrive(res = laml.sig, fdrCutOff = 0.1, useFraction = TRUE)
```

### Adding and summarizing pfam domains
Adds pfam domain information to the amino acid changes
```{r}
laml.pfam = pfamDomains(maf = laml, AACol = 'Protein_Change', top = 10)
```


```{r}
laml.pfam$proteinSummary[,1:7, with = FALSE]
```

```{r}
laml.pfam$domainSummary[,1:3, with = FALSE]
```


### Pan-Cancer comparison
```{r}
#MutsigCV results for TCGA-AML
laml.mutsig <- system.file("extdata", "LAML_sig_genes.txt.gz", package = "maftools")
pancanComparison(mutsigResults = laml.mutsig, qval = 0.1, cohortName = 'LAML', inputSampleSize = 200, label = 1)

```

## Survival analysis
```{r}
#Survival analysis based on grouping of DNMT3A mutation status
mafSurvival(maf = laml, genes = 'DNMT3A', time = 'days_to_last_followup', Status = 'Overall_Survival_Status', isTCGA = TRUE)
```


### Predict genesets associated with survival
Identify set of genes which results in poor survival
```{r}
prog_geneset = survGroup(maf = laml, top = 20, geneSetSize = 2, time = "days_to_last_followup", Status = "Overall_Survival_Status", verbose = FALSE)
```


```{r}
print(prog_geneset)
```

```{r}
mafSurvGroup(maf = laml, geneSet = c("DNMT3A", "FLT3"), time = "days_to_last_followup", Status = "Overall_Survival_Status")
```


```{r}
#Primary APL MAF
primary.apl = system.file("extdata", "APL_primary.maf.gz", package = "maftools")
primary.apl = read.maf(maf = primary.apl)
#Relapse APL MAF
relapse.apl = system.file("extdata", "APL_relapse.maf.gz", package = "maftools")
relapse.apl = read.maf(maf = relapse.apl)
```

### Comparing two cohorts (MAFs)
compare two different type of cohorts to detect differnet mutations.
```{r}
pt.vs.rt <- mafCompare(m1 = primary.apl, m2 = relapse.apl, m1Name = 'Primary', m2Name = 'Relapse', minMut = 5)
print(pt.vs.rt)
```

### Forest plots
```{r}
forestPlot(mafCompareRes = pt.vs.rt, pVal = 0.1, color = c('royalblue', 'maroon'), geneFontSize = 0.8)
```
### Co-onco plots
```{r}
genes = c("PML", "RARA", "RUNX1", "ARID1B", "FLT3")
coOncoplot(m1 = primary.apl, m2 = relapse.apl, m1Name = 'PrimaryAPL', m2Name = 'RelapseAPL', genes = genes, removeNonMutated = TRUE)
```
### Lollipop plot-2s
```{r}
lollipopPlot2(m1 = primary.apl, m2 = relapse.apl, gene = "PML", AACol1 = "amino_acid_change", AACol2 = "amino_acid_change", m1_name = "Primary", m2_name = "Relapse")
```

### Clinical enrichment analysis

clinicalEnrichment is another function which takes any clinical feature associated with the samples and performs enrichment analysis. It performs various groupwise and pairwise comparisions to identify enriched mutations for every category within a clincila feature. Below is an example to identify mutations associated with FAB_classification.
```{r}
fab.ce = clinicalEnrichment(maf = laml, clinicalFeature = 'FAB_classification')
```


```{r}
plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05)
```

### Drug-Gene Interactions

Check the interaction from the Drug Gene Interaction database
```{r}
dgi = drugInteractions(maf = laml, fontSize = 0.75)
```


```{r}
dnmt3a.dgi = drugInteractions(genes = "DNMT3A", drugs = TRUE)
```
### Oncogenic Signaling Pathways

Find the oncogenic signaling pathways in TCGA cohorts.
```{r}
OncogenicPathways(maf = laml)
```


```{r}
PlotOncogenicPathways(maf = laml, pathways = "RTK-RAS")
```

### Mutational Signatures
Each cancer has its own mutation just like our signatures. So that we could make a signature characterized by specific pattern of nucleotide substitutions.


```{r}
library(BSgenome.Hsapiens.UCSC.hg19, quietly = TRUE)
```


```{r}
laml.tnm = trinucleotideMatrix(maf = laml, prefix = 'chr', add = TRUE, ref_genome = "BSgenome.Hsapiens.UCSC.hg19")
```

### Differences between APOBEC enriched and non-enriched samples
```{r}
plotApobecDiff(tnm = laml.tnm, maf = laml, pVal = 0.2)
```

### Signature analysis
Signature analysis includes following steps.

1. estimateSignatures - which runs NMF on a range of values and measures the goodness of fit - in terms of Cophenetic correlation.
2. plotCophenetic - which draws an elblow plot and helps you to decide optimal number of signatures. Best possible signature is the value at which Cophenetic correlation drops significantly.
3. extractSignatures - uses non-negative matrix factorization to decompose the matrix into n signatures. n is chosen based on the above two steps. In case if you already have a good estimate of n, you can skip above two steps.
4. compareSignatures - extracted signatures from above step can be compared to known signatures11 from COSMIC database, and cosine similarity is calculated to identify best match.
plotSignatures - plots signatures
```{r}
library('NMF')
```



```{r}
laml.sign = estimateSignatures(mat = laml.tnm, nTry = 6, pConstant = 0.1)
```


```{r}
plotCophenetic(res = laml.sign)
```

```{r}
laml.sig = extractSignatures(mat = laml.tnm, n = 3, pConstant = 0.1)
```


```{r}
laml.og30.cosm = compareSignatures(nmfRes = laml.sig, sig_db = "legacy")
```



```{r}
#Compate against updated version3 60 signatures 
laml.v3.cosm = compareSignatures(nmfRes = laml.sig, sig_db = "SBS")
```



```{r}
library('pheatmap')
pheatmap::pheatmap(mat = laml.og30.cosm$cosine_similarities, cluster_rows = FALSE, main = "cosine similarity against validated signatures")
```



```{r}
maftools::plotSignatures(nmfRes = laml.sig, title_size = 0.8)
```


### Signature enrichment analysis

Signatures can further be assigned to samples and enrichment analysis can be performd using signatureEnrichment funtion, which identifies mutations enriched in every signature identified.
```{r}
laml.se = signatureEnrichment(maf = laml, sig_res = laml.sig)
```


```{r}
plotEnrichmentResults(enrich_res = laml.se, pVal = 0.05)
```


